molecule/user_provided/prepare.yml

```diff
@@ -0,0 +1,81 @@
+---
+- name: Prepare
+  hosts: localhost
+  connection: local
+  gather_facts: false
+  become: yes
+  tasks:
+    - name: stopping service
+      ansible.builtin.service:
+        name: mariadb
+        state: stopped
+
+    - name: uninstalling
+      ansible.builtin.yum:
+        name:
+          - mariadb
+          - mariadb-server
+          - redis
```

Comment:
I guess I understand that you want to make sure that there is no preexisting initialized mariadb instance that would interfere with the test; however, I'm not sure why you would uninstall the packages, because you can just rely on the install tasks with "state: latest" that you use later to upgrade to latest version. I thing uninstalling is redundant here.